packages/launch/src/hooks/use-domain-selection.ts

```diff
@@ -1,15 +1,62 @@
 /**
  * External dependencies
  */
+import * as React from 'react';
 import { useDispatch, useSelect } from '@wordpress/data';
 import type { DomainSuggestions } from '@automattic/data-stores';
 
 /**
  * Internal dependencies
  */
-import { LAUNCH_STORE } from '../stores';
+import { LAUNCH_STORE, SITE_STORE, DOMAIN_SUGGESTIONS_STORE } from '../stores';
+import LaunchContext from '../context';
+import { isDomainProduct } from '../utils';
+import type { Product, DomainProduct } from '../utils';
 
-export function useDomainSelection() {
+export function useDomainProductFromCart(): DomainProduct | undefined {
+	const { siteId } = React.useContext( LaunchContext );
```

Comment:
About this line, I'm not sure I should retrieve the `siteId` from the `LaunchContext` or from `SITE_STORE` (in terms of reusability of this hook).

Reply:
Getting `siteId ` from context is fine.